askama_shared/src/input.rs

```diff
@@ -59,76 +83,115 @@ impl TemplateInput<'_> {
         let mut syntax = None;
         for item in template_args {
             let pair = match item {
-                syn::NestedMeta::Meta(syn::Meta::NameValue(ref pair)) => pair,
-                _ => {
-                    return Err(format!(
-                        "unsupported attribute argument {:?}",
-                        item.to_token_stream()
-                    )
-                    .into())
+                syn::NestedMeta::Meta(syn::Meta::NameValue(pair)) => pair,
+                meta => {
+                    return Err(syn::Error::new(
+                        meta.span(),
+                        "unsupported attribute argument",
+                    ));
                 }
             };
 
-            if pair.path.is_ident("path") {
+            let ident = match pair.path.get_ident() {
+                Some(ident) => ident,
+                None => continue,
```

Comment:
Unsure if intentional, but the existing could would also trigger the "unsupported attribute key" error in this case

Reply:
A proc_macro is [supposed to ignore](https://doc.rust-lang.org/reference/procedural-macros.html#derive-macro-helper-attributes) "derive macro helper attributes" it didn't define. The reason is that a derive macro is only ever additive, so you see attribute of other proc_macros, too. Maybe you want to build a template using clap::Parser, then Askama should not prevent you from doing so.

So IMO the old implementation was a bug, but fixing it was not my primary intention. :)

Reply:
Ahh makes sense. Just wanted to be sure :)

Reply:
Sorry, I was wrong! I was thinking of line 36: `let ident = match attr.path.get_ident() {`

In here syn's parse_meta() already ensured that `pair.path.get_ident()` cannot be none, so I made this case unreachable!().